src/reducers/console.js

```diff
@@ -26,7 +26,11 @@ export default function console(stateIn, {type, payload, meta}) {
     case 'EVALUATE_CONSOLE_ENTRY':
       return state.set(
         meta.key,
-        new ConsoleEntry({expression: payload}),
+        new ConsoleEntry({expression: payload, status: 'active'}),
+      );
+    case 'DEACTIVATE_CONSOLE_ENTRIES':
```

Comment:
Instead of explicitly deactivating console entries, I think we can calculate whether a given entry should be deactivated based on two pieces of information:

* The key of the current compiled project
* The key of the compiled project that evaluated the entry

With these two pieces of information, we can just built logic right into the presentational components that says: “a console entry is disabled if it was evaluated by a compiled project version that is not the current (tail) compiled project”.

This would mean introducing no new actions at all, and we would no longer need to worry about the question of “what are actions for which we will want to grey the defunct expressions?”

We would of course need to update the `CONSOLE_VALUE_PRODUCED` and `CONSOLE_ERROR_PRODUCED` actions to include the compiled project key, and the reducer to include that information in the entry record. This would have the added advantage of giving us an explicit way to know which entries have been evaluated.